src/Domain/PayPalUrlGenerator.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace WMDE\Fundraising\Frontend\Domain;
+
+/**
+ * @licence GNU GPL v2+
+ * @author Kai Nissen < kai.nissen@wikimedia.de >
+ */
+class PayPalUrlGenerator {
+
+	private $config;
+
+	public function __construct( PayPalConfig $config ) {
+		$this->config = $config;
+	}
+
+	public function generateUrl( int $id, float $amount, int $interval, string $accessToken, string $updateToken ) {
```

Comment:
I had to check the caller to verify this is the donation id, so $donationId is better